services/ux-backend/handlers/onboarding/clienttokens/handler.go

```diff
@@ -66,7 +63,19 @@ func handlePost(w http.ResponseWriter, r *http.Request, tokenLifetimeInHours int
 		return
 	}
 
-	if onboardingToken, err := util.GenerateClientOnboardingToken(tokenLifetimeInHours, onboardingPrivateKeyFilePath, storageQuotaInGiB, storageCluster.UID); err != nil {
+	privateKey, err := util.LoadOnboardingValidationPrivateKey(ctx, cl, namespace)
+	klog.Info("Getting the Pem key")
+	if err != nil {
+		http.Error(w, fmt.Sprintf("Failed to get private key: %v", err), http.StatusBadRequest)
+		return
+	}
```

Comment:
1. It does not help to log after the action, you should log before the action 
2. Why should this context even know that we are using a `pem` key?
3. The error message should be more descriptive and needs to give more context

```suggestion
	klog.Info("Loading onboarding validation private Key")
	if err := privateKey, err := util.LoadOnboardingValidationPrivateKey(ctx, cl, namespace); err != nil {
		http.Error(w, fmt.Sprintf("Failed loading onboarding validation private: %v", err), http.StatusBadRequest)
		return
	}
```